client/python/datajunction/client.py

```diff
@@ -215,7 +216,281 @@ def data(  # pylint: disable=too-many-arguments,too-many-locals
             )
 
 
-class DJWriter(DJReader):
+class DJBuilder(DJClient):
     """
-    Client class to manage all your DJ dag: nodes, attributes, namespaces, tags.
+    Client class for DJ node manipulation.
     """
+
+    #
+    # Catalogs and Engines
+    #
+    def list_catalogs(self):
+        """
+        Gets all catalogs.
+        """
+        response = self._session.get("/catalogs/", timeout=self._timeout)
+        return response.json()
+
+    def list_engines(self):
+        """
+        Gets all engines.
+        """
+        response = self._session.get("/engines/", timeout=self._timeout)
+        return response.json()
+
+    #
+    # Namespace
+    #
+    # TODO: add delete namespace (with warnings) # pylint: disable=fixme
+    def namespace(self, namespace: str) -> "Namespace":
```

Comment:
Should this actually call `create_namespace`? I would've thought that not prefixing something with `create_` would imply that nothing changes server-side, but it will just return an error if the namespace doesn't exist.